main/src/cgeo/geocaching/network/OAuth.java

```diff
@@ -22,8 +22,10 @@ public static void signOAuth(final String host,
                 "oauth_nonce", CryptUtils.md5(Long.toString(System.currentTimeMillis())),
                 "oauth_signature_method", "HMAC-SHA1",
                 "oauth_timestamp", Long.toString(System.currentTimeMillis() / 1000),
-                "oauth_token", StringUtils.defaultString(tokens.getTokenPublic()),
                 "oauth_version", "1.0");
+        if (tokens.getTokenPublic() != null && !tokens.getTokenPublic().isEmpty()) {
+            params.put("oauth_token", StringUtils.defaultString(tokens.getTokenPublic()));
+        }
```

Comment:
This looks suspicious, as the previous implementation was explicitly always setting the oauth_token-parameter, even in the initial request. Scouring the docs, this seems not to be required, but did you cross-check that other authorizations still work (e.g. twitter, oc.*)?

Reply:
I haven't checked with other services but from our side it doesn't work when it sends empty oauth token. It all depends on underlying signing mechanism, maybe on other services it considers empty token.
How should I do it then? Extract this class out of OC and add extra parameter to it?